src/dns.c

```diff
@@ -2379,145 +2495,47 @@ hsk_dns_name_read_size(
 }
 
 bool
-hsk_dns_name_alloc(
-  uint8_t **data,
-  size_t *data_len,
-  const hsk_dns_dmp_t *dmp,
-  char **name
-) {
-  int size = hsk_dns_name_read_size(*data, *data_len, dmp);
-
-  if (size == -1)
-    return false;
-
-  char *n = malloc(size + 1);
-
-  if (!n)
+hsk_dns_name_verify(const uint8_t *name) {
```

Comment:
Name of this method in hnsd is misleading. It's `hsk_dns_tld_verify` instead of `name_verify`. In hsd it's easier to understand rules->nameVerify automatically means `tld` verify. But not in hnsd.

Reply:
> It's worth to check if this needs to be returned back to the methods we removed this as well as check dirty names. Even though dirty names are not a thing, we still need to make sure TLDs are following Handshake rules on top of DNS ones.

Again I think that since we are sanitizing all input with this in `hsk_ns_onrecv()` I don't think we need to put it back anywhere else downstream ...?